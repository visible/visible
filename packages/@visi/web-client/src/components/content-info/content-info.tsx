import React from 'react';

import { Trans, useTranslation } from '../../utils/i18next';
import { Divider, Layout, NavItem, Typography } from '../ui';

type NavColumnProps = {
  title: string;
  items: NavItem[];
};

const NavColumn = ({ title, items }: NavColumnProps) => {
  return (
    <div>
      <Typography variant="h4" fontSize="base" color="wash" className="mb-2">
        {title}
      </Typography>
      <ul className="space-y-1">
        {items.map((navItem, i) => (
          <li key={`item-${i}`}>
            <a
              href={navItem.href}
              target="_blank"
              rel="noreferrer"
              className="hover:underline text-gray-600"
            >
              {navItem.children}
            </a>
          </li>
        ))}
      </ul>
    </div>
  );
};

const Nav = ({ columns }: { columns: NavColumnProps[] }) => {
  return (
    <div className="flex flex-col space-y-2 md:flex-row md:space-x-12 md:space-y-0">
      {columns.map((column, i) => (
        <NavColumn key={`item-${i}`} {...column} />
      ))}
    </div>
  );
};

const Copyright = () => {
  return (
    <div className="flex justify-between space-x-24">
      <div>
        <Typography fontSize="sm" color="wash">
          Copyright 2020 Ryō Igarashi
          <br />
          Distributed under GNU Affero General Public License
        </Typography>
      </div>

      <div>
        <Typography fontSize="sm" color="wash">
          <span>
            Photo by{' '}
            <a href="https://unsplash.com/@sctgrhm?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
              Scott Graham
            </a>{' '}
            on{' '}
            <a href="https://unsplash.com/?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">
              Unsplash
            </a>
          </span>
        </Typography>

        <Typography fontSize="sm" color="wash">
          Icons made by{' '}
          <a href="https://www.flaticon.com/authors/freepik" title="Freepik">
            Freepik
          </a>{' '}
          from{' '}
          <a href="https://www.flaticon.com/" title="Flaticon">
            {' '}
            www.flaticon.com
          </a>
        </Typography>
      </div>
    </div>
  );
};

export const Sponsor = () => {
  const { t } = useTranslation();

  return (
    <div className="flex flex-col items-center w-full space-y-8">
      <Typography variant="h4" fontSize="base" color="wash">
        {t('sponsor.title', 'Sponsored link')}
      </Typography>

      <a
        href="https://vps.sakura.ad.jp/"
        target="_blank"
        rel="noreferrer"
        className="hover:opacity-75 transition-all duration-75"
      >
        <figure className="flex items-center space-x-4">
          <img
            src="/static/sakura-no-vps.png"
            alt={t('sponsor.sakura-no-vps-logo', 'さくらのVPSのロゴ')}
            width="130px"
          />

          <figcaption lang="ja">
            <Typography fontSize="3xl" fontWeight="bold">
              {t('sponsor.sakura-no-vps', 'さくらのVPS')}
            </Typography>
          </figcaption>
        </figure>
      </a>

      <Typography color="wash">
        <Trans i18nKey="sponsor.description">
          This website is powered the infrastructure of{' '}
          <a
            href="https://vps.sakura.ad.jp/"
            target="_blank"
            rel="noreferrer"
            className="text-primary-500 hover:underline"
          >
            さくらのVPS
          </a>{' '}
          by{' '}
          <a
            href="https://www.sakura.ad.jp/"
            target="_blank"
            rel="noreferrer"
            className="text-primary-500 hover:underline"
          >
            SAKURA internet
          </a>
        </Trans>
      </Typography>
    </div>
  );
};

type ContentInfoProps = JSX.IntrinsicElements['footer'];

export const ContentInfo = ({ ref: _ref }: ContentInfoProps) => {
  const { t } = useTranslation();

  const columns = [
    {
      title: t('content-info.contact.title', 'Contact'),
      items: [
        {
          href: 'mailto:n33t5hin@gmail.com',
          children: t('content-info.contact.email', 'Email'),
        },
        {
          href: 'https://twitter.com/visible_hq',
          children: t('content-info.contact.twitter', 'Twitter'),
        },
        {
          href: 'https://github.com/visible/visible/issues/new',
          children: t('content-info.contact.github-issues', 'Report Bugs'),
        },
      ],
    },
    {
      title: t('content-info.resources.title', 'Resources'),
      items: [
        {
          href: 'https://github.com/visible/visible/blob/develop/docs',
          children: t('content-info.resources.docs', 'Documentation'),
        },
        {
          href: 'https://github.com/visible/visible',
          children: t('content-info.resources.github', 'GitHub'),
        },
      ],
    },
    {
      title: t('content-info.product.title', 'Product'),
      items: [
        {
          href: 'http://visi.dev',
          children: t('content-info.product.get-started', 'Get started'),
        },
        {
          href: 'http://eepurl.com/he5d7b',
          children: t('content-info.product.newsletter', 'Newsletter'),
        },
        {
          href:
            'https://github.com/visible/visible/tree/develop/packages/%40visi/web-client/public/static',
          children: t('content-info.product.press-kit', 'Press kit'),
        },
      ],
    },
  ];

  return (
    <Layout.Footer ref={_ref} className="py-6">
      <Layout.Container>
        <Sponsor />
      </Layout.Container>

      <Divider />

      <Layout.Container>
        <div className="w-full space-y-8 mx-auto lg:max-w-screen-md">
          <div className="flex flex-col md:flex-row md:space-x-24 md:justify-between">
            <img src="/static/logo-gray.svg" alt="Visible" width="100px" />
            <Nav columns={columns} />
          </div>
          <Copyright />
        </div>
      </Layout.Container>
    </Layout.Footer>
  );
};
