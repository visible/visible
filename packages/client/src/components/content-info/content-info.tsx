import * as UI from '@visi/ui';
import React from 'react';

type ContentInfoProps = JSX.IntrinsicElements['footer'];

export const ContentInfo = ({ ref: _ref, ...props }: ContentInfoProps) => {
  return (
    <UI.ContentInfo {...props}>
      <UI.List>
        <UI.ListItem>
          <a href="mailto:n33t5hin@gmail.com">Contact</a>
        </UI.ListItem>
        <UI.ListItem>
          <a href="https://github.com/neet/visible">GitHub</a>
        </UI.ListItem>
        <UI.ListItem>
          <a href="https://twitter.com/TheGodOfNeet">Twitter</a>
        </UI.ListItem>
      </UI.List>
    </UI.ContentInfo>
  );
};
