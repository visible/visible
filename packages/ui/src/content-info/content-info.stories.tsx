import React from 'react';
import { List, ListItem } from '../list';
import { ContentInfo } from '.';

export default {
  title: 'ContentInfo',
  component: ContentInfo,
};

export const normal = () => {
  return (
    <ContentInfo>
      <nav>
        <List>
          <ListItem>GitHub</ListItem>
          <ListItem>Twitter</ListItem>
          <ListItem>Email</ListItem>
        </List>
      </nav>
    </ContentInfo>
  );
};
