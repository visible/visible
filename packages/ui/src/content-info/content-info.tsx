// import React from 'react';
import styled from 'styled-components';

import { List } from '../list';

export const ContentInfo = styled.footer`
  transition: 0.15s ease-out;
  opacity: 0.5;
  font-size: 12px;

  a {
    color: ${({ theme }) => theme.foreground.wash};
  }

  ${List} {
    display: flex;
    justify-content: space-around;
    width: 1080px;
    margin: auto;
    padding: 0;
    list-style: none;
  }

  &:hover,
  &:focus {
    transition: 0.15s ease-in;
    opacity: 1;
  }
`;
