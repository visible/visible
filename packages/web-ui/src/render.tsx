import { render as defaultRender } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from './__fixtures__/default-theme';

const TestProvider: React.FC = props => {
  const { children } = props;

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const render = (ui: React.ReactElement) => {
  return defaultRender(ui, { wrapper: TestProvider });
};
