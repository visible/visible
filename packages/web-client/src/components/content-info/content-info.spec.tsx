import { cleanup } from '@testing-library/react';
import React from 'react';

import { render } from '../../tests/render';
import { ContentInfo } from './content-info';

describe('ContentInfo', () => {
  afterEach(() => {
    cleanup();
  });

  it('matches to the snapshot', () => {
    const { container } = render(<ContentInfo />);
    expect(container).toMatchSnapshot();
  });
});
