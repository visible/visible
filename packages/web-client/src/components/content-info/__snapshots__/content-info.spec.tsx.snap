// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentInfo matches to the snapshot 1`] = `
.c2 {
  list-style-type: disc;
}

.c0 {
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  opacity: 0.5;
  font-size: 12px;
}

.c0 a {
  color: #666666;
}

.c0 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 1080px;
  margin: auto;
  padding: 0;
  list-style: none;
}

.c0:hover,
.c0:focus {
  -webkit-transition: 0.15s ease-in;
  transition: 0.15s ease-in;
  opacity: 1;
}

<div>
  <footer
    class="c0"
  >
    <ul
      class="c1 c2"
    >
      <li
        class=""
      >
        <a
          href="mailto:n33t5hin@gmail.com"
        >
          Contact
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="https://github.com/neet/visible"
        >
          GitHub
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="https://twitter.com/TheGodOfNeet"
        >
          Twitter
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
